### PR TITLE
Test signals

### PR DESCRIPTION
**Pull request by sunfishcode**
This adds some basic tests for signal.h functionality added in https://github.com/WebAssembly/wasi-libc/pull/183

### CODE REVIEW

File: tests/general/sigabrt.c [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+int main(void)
+{
```
Comment (sbc100) [inline tests/general/sigabrt.c:6]: The style here seems to be curlys on the same line as the declaration?

File: tests/general/signals.c [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+#endif
+#ifndef SIGINT
+#error SIGINT not defined
+#endif
+#ifndef SIGSEGV
+#error SIGSEGV not defined
+#endif
+#ifndef SIGTERM
+#error SIGTERM not defined
+#endif
```
Comment (sbc100) [inline tests/general/signals.c:26]: Are these really necessary?    Seems like just using them should be fine.

Comment (sunfishcode) [reply tests/general/signals.c]: I've now extended the test to actually use all these.

File: tests/general/sigabrt.c (hunk 1 above)
Comment (sunfishcode) [reply tests/general/sigabrt.c]: Fixed.

**Comment by sunfishcode** [comment]
The CI failure is due to needing a newer wasi-libc; cc #141.

**Comment by sunfishcode** [comment]
CI passed.

**Comment by pchickey** [comment]
the built artifacts from here pass lucet's test suite as well, so we're set to go. I'll pushed the tag, we'll let CI build us the release artifacts now.

**Pull request merged**